adofai-roulette.py

```diff
@@ -2,74 +2,50 @@
 
 # setup song, difficulty, and creator lists
 
-tempfile = open("songs.txt", encoding="utf-8")
-songlistbad = tempfile.readlines()
-tempfile2 = open("difficulties.txt", encoding="utf-8")
-difficultylistbad = tempfile2.readlines()
-tempfile3 = open("creators.txt", encoding = "utf-8")
-creatorlistbad = tempfile3.readlines()
+def readLinesAndStrip(filename: str) -> list[str]:
+    with open(filename, encoding="utf-8") as file:
+        lines = file.readlines()
 
-songlist = []
-difficultylist = []
-creatorlist = []
+    return [line.strip() for line in lines]
 
-for a in songlistbad:
-    songlist.append(a.strip())
-for a in difficultylistbad:
-    difficultylist.append(a.strip())
-for a in creatorlistbad:
-    creatorlist.append(a.strip())
+
+songs = readLinesAndStrip("songs.txt")
+difficulties = readLinesAndStrip("difficulties.txt")
+creators = readLinesAndStrip("creators.txt")
 
 # variable stuff
 
 print("type 'balls' to give up, type your percentage to continue")
 
-response = 0
-oldpercent = 0
+percent = 0
 score = 0
 
-#initial loop
-
-rng = random.randint(1, len(songlist))
-del songlist[rng]
-print(songlist[rng] + " - " + creatorlist[rng] + " - " + difficultylist[rng] + " - Goal: 1%")
-print("")
-while True:
-    response = input("")
-    try:
-        int(response)
-    except ValueError:
-        continue
-    break
-response = int(response)
-oldpercent = response
-print("")
-
-#real loop
+while percent < 100:
+    rng = random.randrange(len(songs))
+    print(f"{songs[rng]} - {creators[rng]} - {difficulties[rng]} - Goal: {percent + 1}%")
+    print()
 
-while response < 100:
-    rng = random.randint(1, len(songlist))
-    print(songlist[rng] + " - " + creatorlist[rng] + " - " + difficultylist[rng] + " - Goal: " + str(response + 1) + "%")
-    print("")
     while True:
         response = input("")

```

Comment:
Note that `response` here is not the same as it was before, it is the response the user gave THIS loop, not what was previously saved.